File: 338-counting-bits/338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] res = new int[n+1];
        for(int i=0;i<=n;i++){
            res[i] = count(i);
        }
        return res;
    }
    
    public int count(int n){
        int count = 0;
        while(n>0){
            count+=1;
            n = n&(n-1);
        }
        return count;
    }
}

File: 338-counting-bits/338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        bitCounts = [0]*(n+1)
        
        powerTwo = 1
        current = 0
        for i in range(1,n+1):
            if i == powerTwo:
                bitCounts[i] = 1
                powerTwo <<= 1
                current = 1
            else:
                bitCounts[i] = bitCounts[current] + 1
                current += 1
        return bitCounts